### PR TITLE
Suggested solrconfig.xml changes for SolrCloud

### PR DESCRIPTION
**Pull request by radu-gheorghe**
I've added some initial settings for:
- commit policy
- transaction log
- update processor chain and handler

I've also removed the replication part (SolrCloud will handle that), and changed the data directory to the default. And changed the directory factory to NRTCachingDirectoryFactory.

### CODE REVIEW

**Comment by banukutlu** [comment]
Closes #249

**Comment by radu-gheorghe** [comment]
Oops, I forgot about RAMBufferSize. Added it now.

File: solr/conf/solrconfig.xml [hunk 1]
```diff
@@ -11,23 +11,63 @@
...
-  <directoryFactory name="DirectoryFactory" class="${solr.directoryFactory:solr.StandardDirectoryFactory}"/>
+  <directoryFactory name="DirectoryFactory" class="${solr.directoryFactory:solr.NRTCachingDirectoryFactory}"/>
 
   <!-- solr lib dirs -->
   <lib dir="${solr.install.dir:../../../..}/contrib/analysis-extras/lib" />
   <lib dir="${solr.install.dir:../../../..}/contrib/analysis-extras/lucene-libs" />
 
-  <dataDir>${solr.blacklight-core.data.dir:}</dataDir>
+  <!-- using default data directory here, for the vanilla Solr installation-->
+  <dataDir>${solr.data.dir:}</dataDir>
```
Comment (banukutlu) [inline solr/conf/solrconfig.xml]: @radu-gheorghe why did we need this change?

**Comment by banukutlu** [comment]
@cdmo travis build fails so I think it will pass after you put your changes in for docker.

**Pull request merged**